modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/CalibrationValue.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.calibration;
+
+import java.util.List;
+
+import com.opengamma.analytics.math.function.Function1D;
+import com.opengamma.analytics.math.matrix.DoubleMatrix1D;
+import com.opengamma.strata.finance.Trade;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+
+/**
+ * Provides the calibration value.
+ * <p>
+ * This function is used during curve calibration.
+ */
+class CalibrationValue
+    extends Function1D<DoubleMatrix1D, DoubleMatrix1D> {
```

Comment:
Do we really have to use `Function1D`? The signature of the `evaluate` method inherited from `Function` is awful. It has a single varargs parameter and throws a runtime exception if there isn't exactly one value.